core/impl/src/main/java/pt/webdetails/cdf/dd/structure/DashboardStructure.java

```diff
@@ -279,6 +279,24 @@ public void newfile( HashMap<String, Object> parameters ) throws Exception {
     }
   }
 
+    /**
+     * Deligates parameters setting to {@link #savesettings(HashMap)} adds updated wcdf to JSON object
+     * and returns it.
+     *
+     * @param parameters
+     * @return JSON object with wcdf that conatins updated parameters
+     * @throws Exception
+     */
+  public JSONObject saveSettingsToWcdf( HashMap<String, Object> parameters ) throws Exception {
+    savesettings( parameters );
+
+    JSONObject wcdfData = this.loadWcdfDescriptor( (String) parameters.get( "file" ) ).toJSON();

```

Comment:
Why not `final JSONObject wcdfData`? Or remove this local variable and pass the value in the `put` function.